### ISSUE TITLE
LoRA fine-tuning destroys pretrained weights — setup_lora called after load_state_dict

### ISSUE THREAD

**Issue by Lgilb3** [fishaudio/fish-speech#1163]
### Self Checks

- [x] This template is only for bug reports. For questions, please visit [Discussions](https://github.com/fishaudio/fish-speech/discussions).
- [x] I have thoroughly reviewed the project documentation (installation, training, inference) but couldn't find information to solve my problem. [English](https://speech.fish.audio/) [中文](https://speech.fish.audio/zh/) [日本語](https://speech.fish.audio/ja/) [Portuguese (Brazil)](https://speech.fish.audio/pt/)
- [x] I have searched for existing issues, including closed ones. [Search issues](https://github.com/fishaudio/fish-speech/issues)
- [x] I confirm that I am using English to submit this report (我已阅读并同意 [Language Policy](https://github.com/fishaudio/fish-speech/issues/515)).
- [x] [FOR CHINESE USERS] 请务必使用英文提交 Issue，否则会被关闭。谢谢！:）
- [x] Please do not modify this template and fill in all required fields.

### Cloud or Self Hosted

Self Hosted (Source)

### Environment Details

Fish Speech: latest main (cloned Feb 17, 2026, commit 53c4ee1)
Model: OpenAudio S1-mini
OS: Windows 11
Python: 3.11
PyTorch: 2.x with CUDA 12.x
GPU: RTX 5070 Ti (16GB)
Self-hosted

### Steps to Reproduce

Clone the repo and set up LoRA fine-tuning per the docs
Run fine-tuning with +lora@model.model.lora_config=r_16_alpha_32
Merge the trained LoRA checkpoint using tools/llama/merge_lora.py
Run inference with the merged model

Or, to verify the bug directly without training:

```

from fish_speech.models.text2semantic.llama import BaseTransformer
from fish_speech.models.text2semantic.lora import LoraConfig

# Load WITHOUT LoRA
model1 = BaseTransformer.from_pretrained(path="checkpoints/openaudio-s1-mini", load_weights=True, lora_config=None)
print(f"Pretrained embeddings std: {model1.embeddings.weight.data.std():.6f}")
# Output: ~0.034
del model1

# Load WITH LoRA
cfg = LoraConfig(r=16, lora_alpha=32, lora_dropout=0.01)
model2 = BaseTransformer.from_pretrained(path="checkpoints/openaudio-s1-mini", load_weights=True, lora_config=cfg)
print(f"After LoRA embeddings std: {model2.embeddings.weight.data.std():.6f}")
# Output: ~1.000 (random!)

```

### ✔️ Expected Behavior

After from_pretrained(load_weights=True, lora_config=...), the model's base weights should be the pretrained weights from model.pth, with LoRA adapters (lora_A, lora_B) initialized to zero contribution. LoRA fine-tuning should learn small corrections on top of the pretrained model.

Actual behavior:

All pretrained weights are destroyed and replaced with random initialization. This happens because commit 781bf1c moved setup_lora() to run AFTER load_state_dict() in from_pretrained(). Since loralib.Linear and loralib.Embedding create new nn.Module objects with random weights in __init__ + reset_parameters(), the pretrained weights loaded in the previous step are discarded.

The result:

Training runs on frozen random base weights instead of pretrained weights
Loss still decreases (embeddings compensate), so training appears to work
Merged models produce gibberish or generic-sounding output instead of the target voice
Training starts at loss ~22 instead of ~8 (where it should start with pretrained weights intact)
Root cause:

In llama.py from_pretrained(), before commit 781bf1c, the order was:

```
model = model_cls(config)
setup_lora(model, lora_config)  # LoRA layers created with random weights
model.load_state_dict(weights)  # Pretrained weights loaded INTO LoRA layers 

```
After 781bf1c, it became:

```
model = model_cls(config)
model.load_state_dict(weights)  # Pretrained weights loaded 
setup_lora(model, lora_config)  # LoRA layers REPLACE loaded weights with random 
```

Fix:

Preserve pretrained weights in setup_lora() by cloning them before creating LoRA layers and restoring them after:

```
def setup_lora(model, lora_config):
    def _replace_embedding(old_embed):
        old_weight = old_embed.weight.data.clone()
        new_embed = lora.Embedding(
            num_embeddings=old_embed.num_embeddings,
            embedding_dim=old_embed.embedding_dim,
            padding_idx=old_embed.padding_idx,
            r=lora_config.r,
            lora_alpha=lora_config.lora_alpha,
        )
        new_embed.weight.data.copy_(old_weight)
        return new_embed

    # ... same pattern for Linear layers:
    # old_weight = old_linear.weight.data.clone()
    # updated_linear = lora.Linear(...)
    # updated_linear.weight.data.copy_(old_weight)
```

This makes setup_lora safe regardless of call order.

### ❌ Actual Behavior

Garbled output
